scrnaseq.Rmd

```diff
@@ -476,23 +618,20 @@ patchwork::wrap_plots(p, ncol=2)
 
 ## Feature plots for known marker genes
 Do cells in individual clusters express provided known marker genes?
-```{r featurePlot_markerGenes, eval=FALSE}
+```{r featurePlot_markerGenes, eval=TRUE}
 g = unique(unlist(param$known_markers))
 g = g[length(g):1]
 p = Seurat::DotPlot(sc, features=g, cols=c("lightgrey", param$col)) + 
   theme(axis.text.x = element_text(angle=90, hjust=1, vjust=.5))
+p = PlotMystyle(p, title="Dot plot of known marker genes")
 p
 
-for (i in 1:length(param$known_markers)) {
-  p = FeaturePlot(sc, features=param$known_markers[[i]], cols=c("lightgrey", param$col))
-  p = ggpubr::annotate_figure(p=p, top=ggpubr::text_grob(label=names(param$known_markers)[i]))
-  print(p)
-}
+# Todo: add feature plots!
 ```
 
 ## Differentially expressed genes
 We next identify genes that are differentially expressed in one cluster compared to all other clusters. Additional gene annotation is added, and the resulting tables are written to file. 
-```{r DEG, results="asis"}
+```{r DEG}
```

Comment:
FindAllMarkers does not work by default on the scaled corrected data. You need to provide the cell cycle variables as covariates in the function.

Reply:
Can you specify this? What exactly did you do?

In this case, I would have to use 
`sc_markers = Seurat::FindAllMarkers(sc, assay="SCT", slot="scale.data", only.pos=FALSE, min.pct=0.25, logfc.threshold=0.25, test.use="MAST")`

or, regress out cell cycle effects in "RNA" data in `ScaleData` as well.

Reply:
Example: `FindAllMarkers(mpeg1_data_cc_removed,assay = "SCT",test.use="LR",latent.vars = c("S.Score","G2M.Score","rep","ribo_pct")`

Reply:
Are you always using `SCT` as assay? I had used `RNA` until now, but would switch now.

Reply:
Another question, is `latent.vars` ("Variables to test") specifying that the scaled data (without cell cycle effects) are to be used?

Reply:
That depends on MAST whether you need real raw countsor can use some corrected counts.

latent.vars gives the covariates/factors that should be blocked in the statistical model. It is like in DESeq2 where you can adjust for Sex or Strain.